wordpress.org/public_html/wp-content/themes/pub/wporg-support/functions.php

```diff
@@ -11,6 +11,36 @@
  */
 add_filter( 'bbp_show_lead_topic', '__return_true' );
 
+/**
+ * Provide a list of local navigation menus.
+ */
+function add_site_navigation_menus( $menus ) {
+	return array(
+		'forums' => array(
+			array(
+				'label' => __( 'Guidelines', 'wporg' ),
+				'url' => '/guidelines/',
+			),
+			array(
+				'label' => __( 'Welcome to Support', 'wporg' ),
+				'url' => '/welcome/',
+			),
+			array(
+				'label' => __( 'Get Involved', 'wporg' ),
+				'url' => '/handbook/',
+			)
+		),
+	);
+}
+add_filter( 'wporg_block_navigation_menus', '\add_site_navigation_menus' );
+
+function register_patterns() {
+	require get_template_directory() . '/patterns/local-nav.php';
+	require get_template_directory() . '/patterns/forums-homepage-header.php';
+	require get_template_directory() . '/patterns/search-field.php';
+}
+add_action( 'init', 'register_patterns' );
```

Comment:
We can manually register patterns, there might be a better way of doing this.

Reply:
Hm… I thought patterns should be loaded from the `patterns` directory in classic themes just like they are in block themes. I tried it (I'm sure you did too), but it's not working here, I have no idea why. Core should be doing exactly what you're doing here.

Even when I get around the cached pattern issue, the pattern appears to register, but does not show up where it should.

For dev-ex, I wonder if we should load all patterns in a single file (for example, [how twentynineteen does it](https://core.trac.wordpress.org/browser/trunk/src/wp-content/themes/twentynineteen/inc/block-patterns.php)), since the patterns dir isn't working as expected? (though ideally we could fix whatever's wrong there…)

Reply:
Added a [DirectoryIterator](https://github.com/WordPress/wordpress.org/pull/201/commits/a385929554dee7ccecda8a2befa5ed95df958019#diff-2d167406fe0e1ea03098dc0c90a59782d8c8d97096d5173788c1052ac82d85ecR41) to require all the files in `/patterns/`, see above